composition.go

```diff
@@ -0,0 +1,110 @@
+package ep
+
+import (
+	"context"
+)
+
+var _ = registerGob(&composition{}, &composeProject{})
+
+// Function is a function that transforms Datasets. Any function that accepts a
+// Dataset and returns a Dataset with the same length is a Function.
```

Comment:
```suggestion
// Function must return a Dataset immediately without waiting for next batch 
```
we might want to compose also function that returns less/more results. the key point here INHO is that function should not be blocking or wait for next batch.